specification/storageactions/StorageAction.Management/models.tsp

```diff
@@ -0,0 +1,594 @@
+  properties?: StorageTaskUpdateProperties;
+}
+
+/**
+ * Properties of the storage task.
+ */
+model StorageTaskUpdateProperties {
+  /**
+   * Storage task version.
+   */
+  @visibility(Lifecycle.Read)
+  @maxValue(1)
+  @minValue(1)
+  taskVersion?: int64;
+
+  /**
+   * Storage Task is enabled when set to true and disabled when set to false
+   */
+  enabled?: boolean;
+
+  /**
+   * Text that describes the purpose of the storage task
+   */
+  description?: string;
+
+  /**
+   * The storage task action that is executed
+   */
+  action?: StorageTaskAction;
+
+  /**
+   * Represents the provisioning state of the storage task.
+   */
+  @visibility(Lifecycle.Read)
+  provisioningState?: ProvisioningState;
+
+  /**
+   * The creation date and time of the storage task in UTC.
+   */
+  @visibility(Lifecycle.Read)
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  creationTimeInUtc?: utcDateTime;
+}
+
+/**
+ * Fetch Storage Tasks Run Summary.
+ */
+model StorageTaskReportSummary is Azure.Core.Page<StorageTaskReportInstance>;
+
+/**
+ * Storage Tasks run report instance
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model StorageTaskReportInstance
+  extends Azure.ResourceManager.Foundations.ProxyResource {
+  /**
+   * Storage task execution report for a run instance.
+   */
+  properties?: StorageTaskReportProperties;
+}
+
+/**
+ * Storage task execution report for a run instance.
+ */
+model StorageTaskReportProperties {
+  /**
+   * Resource ID of the Storage Task Assignment associated with this reported run.
+   */
+  @visibility(Lifecycle.Read)
+  taskAssignmentId?: Azure.Core.armResourceIdentifier<[
+    {
+      type: "Microsoft.Storage/storageAccounts/storageTaskAssignments";
+    }
+  ]>;
+
+  /**
+   * Resource ID of the Storage Account where this reported run executed.
+   */
+  @visibility(Lifecycle.Read)
+  storageAccountId?: Azure.Core.armResourceIdentifier<[
+    {
+      type: "Microsoft.Storage/storageAccounts";
+    }
+  ]>;
+
+  /**
+   * Start time of the run instance. Filter options such as startTime gt '2023-06-26T20:51:24.4494016Z' and other comparison operators can be used as described for DateTime properties in https://learn.microsoft.com/en-us/rest/api/storageservices/querying-tables-and-entities#supported-comparison-operators
+   */
+  @visibility(Lifecycle.Read)
+  startTime?: string;
+
+  /**
+   * End time of the run instance. Filter options such as startTime gt '2023-06-26T20:51:24.4494016Z' and other comparison operators can be used as described for DateTime properties in https://learn.microsoft.com/en-us/rest/api/storageservices/querying-tables-and-entities#supported-comparison-operators
+   */
+  @visibility(Lifecycle.Read)
+  finishTime?: string;
+
+  /**
```

Comment:
should these be modeled as datetimes?

Reply:
It's string before.

Reply:
The goal is to keep the swagger equivalent here, as service as already GA